selenium_test.py

```diff
@@ -0,0 +1,222 @@
+        return [keyword.strip() for keyword in fd.readlines()]
+
+
+def economist_scraper(keyword, startyear, endyear):
+    driver = webdriver.Firefox(executable_path='/Users/dylanedwards/PycharmProjects/economist-metasearcher/geckodriver')
+    driver.get("http://www.economist.com/historicalarchive")
+
+    # tests to make sure we are on login page
+    assert "Economist" in driver.title
+
+    # inputs username
+    username = driver.find_element_by_name("email")
+    username.clear()
+    username.send_keys("dylanedwards290@gmail.com")
+
+    # inputs passwords
+    password = driver.find_element_by_name("password")
+    password.clear()
+    password.send_keys("duDnis-0givza-carfed")
+
+    # clicks log in button
+    logIn = driver.find_element_by_xpath("/html/body/form/div[2]/table/tbody/tr[5]/td[2]/a")
+    logIn.click()
+
+    # waits for the presence of the search input area on new form page before running
+    searchWait = WebDriverWait(driver, 10).until(
+        EC.presence_of_element_located((By.ID, "inputFieldValue_0"))
+    )
+
+    # inputs search term
+    searchTerm = driver.find_element_by_xpath("// *[ @ id = 'inputFieldValue_0']")
+    searchTerm.clear()
+    searchTerm.send_keys(keyword)
+
+    # selects keyword radio button
+    keywordButton = driver.find_element_by_xpath("/html/body/div[3]/form/div[1]/div[3]/input[2]")
+    keywordButton.click()
+
+    # select between date from drop down
+    Between = Select(driver.find_element_by_xpath('//*[@id="dateLimiterValue_da_dateMode_0"]'))
+    Between.select_by_value('4')  # select by value
+
+    # select start day from drop down
+    startDay = Select(driver.find_element_by_xpath('// *[ @ id = "dateLimiterValue_da_fromDay"]'))
+    startDay.select_by_value('01')  # select by value
+
+    # select start month from drop down
+    startMonth = Select(driver.find_element_by_xpath('// *[ @ id = "dateLimiterValue_da_fromMonth"]'))
+    startMonth.select_by_value('01')  # select by value
+
+    # select start Year from drop down
+    startYear = Select(driver.find_element_by_xpath('// *[ @ id = "dateLimiterValue_da_fromYear"]'))
+    startYear.select_by_value(startyear)  # select by value
+
+    # select end day from drop down
+    endDay = Select(driver.find_element_by_xpath('//*[@id="dateLimiterValue_da_toDay"]'))
+    endDay.select_by_value('01')  # select by value
+
+    # select end month from drop down
+    endMonth = Select(driver.find_element_by_xpath('//*[@id="dateLimiterValue_da_toMonth"]'))
+    endMonth.select_by_value('01')  # select by value
+
+    # select end Year from drop down
+    endYear = Select(driver.find_element_by_xpath('//*[@id="dateLimiterValue_da_toYear"]'))
+    endYear.select_by_value(endyear)  # select by value
+
+    # clicks Search Submit button
+    searchSubmit = driver.find_element_by_xpath("/html/body/div[3]/form/div[1]/div[2]/a")
+    searchSubmit.click()
+
+    # calls collect results on search results
+    collectResults(driver, keyword)
```

Comment:
`collectResults()` should return its own list of `Link`s or `Article`s, which should be added to your list here.